src/main/java/picard/vcf/MakeVcfSampleNameMap.java

```diff
@@ -0,0 +1,134 @@
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+
+/**
+ * Creates a TSV from sample name to VCF/GVCF path, with one line per input.
+ *
+ * <h3>Summary</h3>
+ * Creates a TSV from sample name to VCF/GVCF path, with one line per input.
+ * Input VCF/GVCFs must contain a header describing exactly one sample.
+ * <br/>
+ * <h3>Usage example:</h3>
+ * <pre>
+ *     java -jar picard.jar MakeVcfSampleNameMap \
+ *      INPUT=sample1.vcf.gz \
+ *      INPUT=sample2.vcf.gz \
+ *      OUTPUT=cohort.sample_map
+ * </pre>
+ *
+ * @author Daniel Moran
+ */
+@CommandLineProgramProperties(
+        summary = MakeVcfSampleNameMap.SUMMARY,
+        oneLineSummary = MakeVcfSampleNameMap.SHORT_SUMMARY,
+        programGroup = VariantManipulationProgramGroup.class
+)
+public class MakeVcfSampleNameMap extends CommandLineProgram {
+
+    static final String SHORT_SUMMARY = "Creates a TSV from sample name to VCF/GVCF path, with one line per input.";
+    static final String SUMMARY = SHORT_SUMMARY + "\n" +
+            "Input VCF/GVCFs must contain a header describing exactly one sample.\n" +
+            "<h4>Usage example:</h4>" +
+            "<pre>" +
+            "    java -jar picard.jar MakeVcfSampleNameMap \\\n" +
+            "        INPUT=sample1.vcf.gz \\\n" +
+            "        INPUT=sample2.vcf.gz \\\n" +
+            "        OUTPUT=cohort.sample_map" +
+            "</pre>";
+
+    @Argument(shortName = StandardOptionDefinitions.INPUT_SHORT_NAME,
+            doc = "One or more input VCFs to extract sample names from.", minElements = 1)
+    public List<String> INPUT;
+
+    @Argument(shortName = StandardOptionDefinitions.OUTPUT_SHORT_NAME, doc = "Output file to write the sample-name map to.")
+    public File OUTPUT;
+
+    @Override
+    protected int doWork() {
+        IOUtil.assertFileIsWritable(OUTPUT);
+
+        final Map<String, String> nameMap = new HashMap<>(INPUT.size());
+
+        for (final String variantPathString : INPUT) {
+            final Path variantPath = getVariantPath(variantPathString);
+            final VCFHeader header = getHeaderFromPath(variantPath);
+
+            final int numberOfSamples = header.getNGenotypeSamples();
+            if (numberOfSamples != 1) {
+                throw new PicardException("Input: " + variantPathString + " was expected to contain a single sample" +
+                        " but actually contained " + numberOfSamples + " samples.");
+            }
+
+            final String sampleName = header.getGenotypeSamples().get(0);
+            final String previousPath = nameMap.put(sampleName, variantPathString);
+            if (previousPath != null) {
+                throw new PicardException("Duplicate sample: " + sampleName + ". Sample was found in both " +
```

Comment:
this cannot be an error...unfortunately. Perhaps just emit a warning? (also, you'll need to switch around your map...

Reply:
Done